packages/subscription-controller/src/types.ts

```diff
@@ -1,49 +1,55 @@
 import type { Hex } from '@metamask/utils';
 
-export enum ProductType {
-  SHIELD = 'shield',
-}
+export const PRODUCT_TYPES = {
+  SHIELD: 'shield',
+} as const;
 
-/** only usd for now */
-export type Currency = 'usd';
+export type ProductType = (typeof PRODUCT_TYPES)[keyof typeof PRODUCT_TYPES];
 
-export type Product = {
-  name: ProductType;
-  id: string;
-  currency: Currency;
-  amount: number;
-};
+export const PAYMENT_TYPES = {
+  byCard: 'card',
+  byCrypto: 'crypto',
+} as const;
 
-export enum PaymentType {
-  byCard = 'card',
-  byCrypto = 'crypto',
-}
+export type PaymentType = (typeof PAYMENT_TYPES)[keyof typeof PAYMENT_TYPES];
```

Comment:
You can simplify this to:
`export type PaymentType = 'card' | 'crypto';`
Same for the other enum style type definitions.

Or is there a benefit of writing it the way you did that I don't see?